src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets

```diff
@@ -50,6 +51,21 @@ Copyright (c) .NET Foundation. All rights reserved.
     </CreateWindowsSdkKnownFrameworkReferences>
   </Target>
 
+  <PropertyGroup Condition="'$(EnablePrunePackageReferences)' == ''">
+    <EnablePrunePackageReferences>true</EnablePrunePackageReferences>
+  </PropertyGroup>
+  
+  <Target Name="AddPrunePackageReferences" BeforeTargets="CollectPrunePackageReferences"
+          Condition="'$(EnablePrunePackageReferences)' == 'true'">
+
+    <GetPackagesToPrune TargetFrameworkIdentifier="$(TargetFrameworkIdentifier)"
+                        TargetFrameworkVersion="$(_TargetFrameworkVersionWithoutV)"
+                        FrameworkReferences="@(FrameworkReference)">
```

Comment:
Do we make sure that the framework references being pruned are only direct ones?

Reply:
This is called before restore, so indirect framework references aren't even available yet.  So yes, these will only be direct framework references.